tests/components/ruckus_unleashed/test_coordinator.py

```diff
@@ -0,0 +1,42 @@
+"""Test the Ruckus Unleashed DataUpdateCoordinator flow."""
+from pyruckus import Ruckus
+
+from homeassistant.components.ruckus_unleashed import (
+    API_MAC,
+    RuckusUnleashedDataUpdateCoordinator,
+)
+from homeassistant.components.ruckus_unleashed.const import (
+    API_CLIENTS,
+    API_CURRENT_ACTIVE_CLIENTS,
+    API_NAME,
+)
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+
+from tests.async_mock import patch
+from tests.components.ruckus_unleashed import CONFIG, TEST_CLIENT
+
+
+async def test_fetch_clients(hass):
+    """Test we can fetch and transform clients."""
+    with patch(
+        "homeassistant.components.ruckus_unleashed.Ruckus.connect",
+        return_value=None,
+    ), patch(
+        "homeassistant.components.ruckus_unleashed.Ruckus.current_active_clients",
+        return_value={API_CURRENT_ACTIVE_CLIENTS: {API_CLIENTS: [TEST_CLIENT]}},
+    ):
+        ruckus = await hass.async_add_executor_job(
+            Ruckus,
+            CONFIG[CONF_HOST],
+            CONFIG[CONF_USERNAME],
+            CONFIG[CONF_PASSWORD],
+        )
+
+        coordinator = RuckusUnleashedDataUpdateCoordinator(hass, ruckus=ruckus)
+
+        await coordinator.async_refresh()
+
+    assert coordinator.data[API_CLIENTS]
```

Comment:
Please don't assert integration details. We can remove this test.

Reply:
I wasn't meaning to assert integration details here.
I meant to test that the coordinator correctly fetches the data from the external library and modifies the returned list into a dict with MAC address as the keys. I saw this done in other coordinators, but if it shouldn't be done here then I can move that functionality back to the external library. Just trying to simplify where I can! Thank you!

https://github.com/home-assistant/core/pull/41824/files#diff-e52d06101b1fab6d4522620cd6968eb02a600764dd1ffc3dc92a6ab88dd2a634R38-R43

Reply:
Yes, that's a detail of the integration which we don't want to test. We just care about the core states. See:
https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations

Reply:
Oh oops. I had definitely seen that before, but I got so caught up in wanting 100% test coverage that I forgot about that...
Fixed. Thank you!